### PR TITLE
[Modules] Added location output to templates

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Added location output to templates (for those where the primary resource that support it)
- Added test case for location output
- Split original outputs tests into separate test cases
- Improved function import for tests

| Pipeline |
| - |
| [AnalysisServices: Servers](https://github.com/Azure/ResourceModules/actions/workflows/ms.analysisservices.servers.yml)|
|[Resources: ResourceGroups](https://github.com/Azure/ResourceModules/actions/workflows/ms.resources.resourcegroups.yml)|
|[Authorization: PolicyAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policyassignments.yml)|
|[Authorization: RoleAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.roleassignments.yml)|

# Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

### CODE REVIEW

File: arm/Microsoft.ContainerService/managedClusters/readme.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ This module deploys Azure Kubernetes Cluster (AKS).
 | `Microsoft.Authorization/roleAssignments` | [2020-10-01-preview](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Authorization/2020-10-01-preview/roleAssignments) |
 | `Microsoft.ContainerService/managedClusters` | [2022-02-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.ContainerService/2022-02-01/managedClusters) |
 | `Microsoft.ContainerService/managedClusters/agentPools` | [2021-08-01](https://docs.microsoft.com/en-us/azure/templates/Microsoft.ContainerService/2021-08-01/managedClusters/agentPools) |
-| `Microsoft.Insights/diagnosticSettings` | [2021-05-01-preview](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Insights/2021-05-01-preview/diagnosticSettings) |
+| `Microsoft.Insights/diagnosticSettings` | [2021-05-01-preview](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Insights/diagnosticSettings) |
```
Comment (eriqua) [inline arm/Microsoft.ContainerService/managedClusters/readme.md:19]: Please revert this change

File: utilities/pipelines/sharedScripts/Get-ScopeOfTemplateFile.ps1 [hunk 2]
```diff
@@ -47,7 +47,7 @@ function Get-ScopeOfTemplateFile {
             Default { throw "[$armSchema] is a non-supported ARM template schema" }
         }
     }
-    Write-Verbose "Determined deployment scope [$deploymentScope]" -Verbose
+    Write-Verbose "Determined deployment scope [$deploymentScope]"
```
Comment (eriqua) [inline utilities/pipelines/sharedScripts/Get-ScopeOfTemplateFile.ps1:50]: Is this intended?

File: arm/Microsoft.ContainerService/managedClusters/readme.md (hunk 1 above)
Comment (eriqua) [inline arm/Microsoft.ContainerService/managedClusters/readme.md:19]: Please revert
```suggestion
| `Microsoft.Insights/diagnosticSettings` | [2021-05-01-preview](https://docs.microsoft.com/en-us/azure/templates/Microsoft.Insights/2021-05-01-preview/diagnosticSettings) |
```

**Review by eriqua — CHANGES_REQUESTED** [review]
Please update wiki pages:

Module design>Outputs section
(if useful) Static validation currently just saying "the minimum set of outputs is returned"

File: utilities/pipelines/sharedScripts/Get-ScopeOfTemplateFile.ps1 (hunk 2 above)
Comment (MrMCake) [reply utilities/pipelines/sharedScripts/Get-ScopeOfTemplateFile.ps1:50]: Yes. Want to remove the verbosity

**Comment by MrMCake** [comment]
> Please update wiki pages:
> 
> Module design>Outputs section (if useful) Static validation currently just saying "the minimum set of outputs is returned"

Good catch. Updated
